.github/workflows/pr-ci.yml

```diff
@@ -1,85 +1,158 @@
 name: CI build and push (PR)
 
 concurrency:
-  group: pr-${{ github.event.number }}
+  group: ci-${{ github.run_id }}
   cancel-in-progress: true
 
 on:
-  pull_request_target:
+  issue_comment:
     types:
-      - opened
-      - reopened
-      - synchronize
-      - labeled
-      - unlabeled
-    branches:
-      - main
-      - v[0-9]+
-      - v[0-9]+.[0-9]+
-      - cryostat-v[0-9]+.[0-9]+
+      - created
 
 jobs:
+  check-before-build:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'cryostatio' && github.event.issue.pull_request && startsWith(github.event.comment.body, '/build_test')
+    steps:
+    - name: Fail if needs-triage label applied
+      if: ${{ contains(github.event.issue.labels.*.name, 'needs-triage') }}
+      run: exit 1
+    - name: Fail if commentor is not a member of the repo
+      if: |
+        github.event.comment.author_association != 'MEMBER' && 
+        github.event.comment.author_association != 'OWNER'
+      uses: thollander/actions-comment-pull-request@v1
+      with:
+        message: |-
+          You do not have permission to run the /build_test command. Please ask a member of the cryostatio 
+          repository to run it for you.
+    - name: React to comment
+      uses: actions/github-script@v4
+      with:
+        script: |
+            const {owner, repo} = context.issue
+            github.reactions.createForIssueComment({
+              owner,
+              repo,
+              comment_id: context.payload.comment.id,
+              content: "+1",
+            });  
+
+  checkout-branch: 
+    runs-on: ubuntu-latest
+    needs: [check-before-build]
+    permissions:
+      contents: read
+      issues: read
+      pull-requests: read
+    outputs: 
+      PR_head_ref: ${{ fromJSON(steps.comment-branch.outputs.result).head.ref }}
+      PR_head_sha: ${{ fromJSON(steps.comment-branch.outputs.result).head.sha }}
+      PR_repo: ${{ fromJSON(steps.comment-branch.outputs.result).head.repo.full_name }}
+      PR_repo_owner: ${{ fromJSON(steps.comment-branch.outputs.result).head.repo.owner.login }}
+    if: github.repository_owner == 'cryostatio' && github.event.issue.pull_request && startsWith(github.event.comment.body, '/build_test')
+    steps:
+    - uses: actions/github-script@v3
+      id: comment-branch
+      with:
+        script: |
+          const request = {
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            pull_number: context.issue.number
+          }
+          core.info(`Getting PR #${request.pull_number} from ${request.owner}/${request.repo}`)
+          try {
+            const result = await github.pulls.get(request)
+            return result.data
```

Comment:
Can we extract the field into a simplified object and return it? This way, the parsing of `fromJSON` should be lighter (i.e. the full original result object is quite large).